lib/perl/Genome/Model/Tools/Analysis/Coverage/CoveragePlot.R

```diff
@@ -0,0 +1,1549 @@
+    width = unit(1.0, "npc") - left - default;
+    height = unit(1.0, "npc") - bottom - default - unit(lines, "lines");
+    
+    # creates a viewport for a plot region
+    vp <- viewport(
+        x = left,
+        y = bottom,
+        width = width,
+        height = height,
+        xscale = cs_scale_scope(xscale, factor=xfactor),
+        yscale = cs_scale_scope(yscale, factor=yfactor),
+        just = c("left", "bottom"),
+        #clip = TRUE,
+
+        name = name
+    );
+    
+    # for debug
+    #str(vp);
+    #print(paste("x=", vp$x));
+    #print(paste("y=", vp$y));
+    
+    return(vp);
+}
+
+
+
+cs_base <- function(title=NULL, lines=2, xlab=NULL, xmargin=2, ylab=NULL, ymargin=2, xscale=NULL, xfactor=0.05, xvisible=T, xrot=0, xtick=-1, xspace=-1, yscale=NULL, yfactor=0.05, yvisible=T, yrot=0, ytick=-1, yspace=-1, default = unit(1.5, "lines"), gp=gpar(), name=NULL, parent=NULL, vp=NULL, xmar = NULL, ymar = NULL)
+{
+    # the chracter identifiers for the grid components
+    if (is.null(name)) nv <- "cs_base" else nv = name;        # the grid viewport
+    nt = "title";  # main title
+    nx = "xaxis";   # x axis
+    ny = "yaxis";   # y axis
+    nxl = "xlabel"; # x label
+    nyl = "ylabel"; # x label
+    nb = "border";   # border
+    
+    # checks the x and y scale for a plot
+    if(is.null(xscale)) stop("xscale not defined");
+    if(! is.vector(xscale, mode="numeric")) stop("wrong data type for xscale");
+    if(is.null(yscale)) stop("yscale not defined");
+    if(! is.vector(yscale, mode="numeric")) stop("wrong data type for yscale");
+    
+    # creates and sets up a viewport for a plot
+    if (is.null(vp))
+    {
+        # sets up the margin of a plot region as a horizontal space from the left
+        xmar <- cs_hmargin(yscale, visible=yvisible, rot=yrot, label=(! is.null(ylab)), default=default);
+        
+        # sets up the margin of a plot region as a vertical space from the bottom
+        ymar <- cs_vmargin(xscale, visible=xvisible, rot=xrot, label=(! is.null(xlab)), default=default);
+        
+        # the lines for a title
+        if (is.null(title)) titlelines <- 0
+        else                titlelines <- lines;
+        
+        # creates a viewport
+        # old line: cv <- cs_viewport2(left=xmar, bottom=ymar, xscale=cs_range(xscale), xfactor=xfactor, yscale=cs_range(yscale), yfactor=yfactor, lines=titlelines, default=default, name=nv);
+        cv <- cs_viewport2(left=xmar, bottom=ymar, xscale=xscale, xfactor=xfactor, yscale=yscale, yfactor=yfactor, lines=titlelines, default=default, name=nv);
+    }
+    else
+    {
+        # uses the given viewport
+        cv <- vp;
+        
+        # the margins obtained from the given viewport
+        # The following margins do not work in an accurate way
+        #xmar = vp$x;
+        #ymar = vp$y;
+        
+        # for debug
+        #print(xmar);
+        #print(vp$x);
+        #print(ymar);
+        #print(vp$y);

```

Comment:
Those comments were already cleaned up. I also realized some functions 
are unnecessary (never used) in the CoveragePlot code. So, I had updated 
the codes accordingly, but I don't know how to push them up onto the 
previous commitment (which you are now seeing). Let me know how to do by 
commands.


On 3/4/2015 11:57 AM, Chris Miller wrote:
>
> In lib/perl/Genome/Model/Tools/Analysis/Coverage/CoveragePlot.R 
> <https://github.com/genome/genome/pull/525#discussion_r25796352>:
>
> > +    }
> > +    else
> > +    {
> > +        # uses the given viewport
> > +        cv <- vp;
> > +
> > +        # the margins obtained from the given viewport
> > +        # The following margins do not work in an accurate way
> > +        #xmar = vp$x;
> > +        #ymar = vp$y;
> > +
> > +        # for debug
> > +        #print(xmar);
> > +        #print(vp$x);
> > +        #print(ymar);
> > +        #print(vp$y);
>
> Lots of comments that could be removed. Debugging statements should be 
> taken out. If the above chunk that's labeled "The following margins do 
> not work in an accurate way" is old code that will not be used, just 
> remove it.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/genome/genome/pull/525/files#r25796352>.
>



____
This email message is a private communication. The information transmitted, including attachments, is intended only for the person or entity to which it is addressed and may contain confidential, privileged, and/or proprietary material. Any review, duplication, retransmission, distribution, or other use of, or taking of any action in reliance upon, this information by persons or entities other than the intended recipient is unauthorized by the sender and is prohibited. If you have received this message in error, please contact the sender immediately by return email and delete the original message from all computer systems. Thank you.